### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by khomaaa**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -12,29 +16,76 @@ const (
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
 	// ваш код здесь
+	if size <= 0 {
+		return []int{}
+	}
+
+	data := make([]int, size)
+	for i := range data {
+		data[i] = rand.Intn(math.MaxInt64)
```
Comment (dirtymew) [inline main.go:25]: rand.Int чем не подошел?

File: main.go [hunk 2]
```diff
@@ -12,29 +16,76 @@ const (
...
+	}
+	return data
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
 	// ваш код здесь
+	max := data[0]
```
Comment (dirtymew) [inline main.go:33]: хорошо бы добавить проверки на пустой массив иначе можно выйти за границы

File: main.go [hunk 3]
```diff
@@ -12,29 +16,76 @@ const (
...
+		start := i * chunkSize
+		end := start + chunkSize
+		if i == CHUNKS-1 {
+			end = len(data)
+		}
+		// launching goroutines
+		go func(index int, chunk []int) {
+			defer wg.Done()
+			localMax := maximum(chunk)
+			maxSlice[index] = localMax
```
Comment (dirtymew) [inline main.go:61]: maxSlice[index] = maximum(chunk)

File: main_test.go [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+}
+
+// TestMaximum according to comments, corrected: made a table test and edge cases
+func TestMaximum(t *testing.T) {
+	tests := []struct {
+		name     string
+		data     []int
+		expected int
+	}{
+		{"One element", []int{42}, 42},
```
Comment (dirtymew) [inline main_test.go:38]: стоило добавить кейсты на nil и пусто массив

**Review by dirtymew — APPROVED** [review]
Привет 

хорошая работа - но есть что улучшить см PR
